modules/data.remote/R/soilgrids.R

```diff
@@ -0,0 +1,224 @@
+##' dbname='betydb', driver=RPostgres::Postgres(),write=FALSE)
+##' 
+##' con <- DBI::dbConnect(drv=bety$driver, dbname=bety$dbname, host=bety$host, 
+##' password=bety$password, user=bety$user)
+##' 
+##' suppressWarnings(site_qry <- glue::glue_sql("SELECT *, ST_X(ST_CENTROID(geometry)) AS lon,
+##' ST_Y(ST_CENTROID(geometry)) AS lat FROM sites WHERE id IN ({ids*})",
+##' ids = c("676","622","678","766","764"), .con = con))
+##' 
+##' suppressWarnings(qry_results.1 <- DBI::dbSendQuery(con,site_qry))
+##' suppressWarnings(qry_results.2 <- DBI::dbFetch(qry_results.1))
+##' DBI::dbClearResult(qry_results.1)
+##' dbDisconnect(con)
+##' 
+##' site_info <- qry_results.2
+##' verbose <- TRUE
+##' system.time(result_soc <- soilgrids.soc.extract(site_info=site_info, verbose=verbose))
+##' result_soc
+##' 
+##' }
+##' 
+##' @importFrom reshape2 melt
+##' @importFrom terra vect project rast extract
+##' @importFrom stats qgamma optim
+##' @importFrom utils flush.console setTxtProgressBar txtProgressBar
+##' 
+##' @return list of two dataframes containing the mean SOC values per location and depth
+##' and the corresponding standard deviation values (uncertainties) for each location 
+##' and depth. Output column names are c("site_id","lat","lon","0-5cm","5-15cm","15-30cm",
+##' "30-60cm","60-100cm","100-200cm")
+##' 
+##' @export
+##' @author Qianyu Li, Shawn P. Serbin
+##' 
+soilgrids.soc.extract <- function (site_info=NULL, outdir=NULL, verbose=TRUE) {
+  
+  
+  if (is.null(site_info)) {
+    stop("Please provide a BETY DB site list containing at least the site id and PostGIS geometry\
+    as lon and lat")
+  }
+  
+  # prepare site info for extraction
+  internal_site_info <- data.frame(site_info$id, site_info$sitename, site_info$lat, 
+                                   site_info$lon, site_info$time_zone)
+  
+  i <- 1 # setup counter
+  #create a variable to store mean and quantile of SOC for each soil depth
+  socquant <- matrix(NA, nrow = 6, ncol = length(internal_site_info$site_info.lon) * 4)
+  siteids <- matrix(NA, nrow = 6, ncol = length(internal_site_info$site_info.id) * 4)
+  lonlat <- cbind(internal_site_info$site_info.lon, internal_site_info$site_info.lat)
+  base_data_url <- "/vsicurl?max_retry=30&retry_delay=60&list_dir=no&url=https://files.isric.org/soilgrids/latest/data/soc/soc_"
+  depths <- c("0-5cm", "5-15cm", "15-30cm", "30-60cm", "60-100cm", "100-200cm")
+  
+  # reproject locations to soilgrids projection
+  #Soilgrids data is using Homolosine projection https://www.isric.org/explore/soilgrids/faq-soilgrids
+  p <- terra::vect(lonlat, crs = "+proj=longlat +datum=WGS84") # Users need to provide lon/lat
+  newcrs <- "+proj=igh +datum=WGS84 +no_defs +towgs84=0,0,0" 
+  p_reproj <- terra::project(p, newcrs) # Transform the point vector to data with Homolosine projection
+  
+  # setup progress bar
+  if (verbose) {
+    j <- 1
+    pb <- txtProgressBar(min = 0, max = length(depths), char="*", width=70, style = 3)
```

Comment:
package?

Reply:
updating to all being [package]::[function]